### PR TITLE
Rewrite LootManager to use a standard Loot Table. Closes #3927

### LINKED ISSUE #3927: EnderIO causes GSON to throw an exception when certain loot tables are serialised
#### Issue Description:

When LootTweaker attempts to serialise certain loot tables(to dump them to a folder), GSON throws an error. This error is not caused by LootTweaker or CraftTweaker, as it only occurs with EnderIO installed. EnderCore is not the cause of the problem. LootTweaker successfully dumps loot tables modified by other mods, such as Actually Additions and BetterBeginnings.

#### What happens:

When LootTweaker attempts to serialise certain loot tables with EnderIO installed, GSON throws an error.

#### What you expected to happen:

LootTweaker to successfully dump all loot tables.

#### Steps to reproduce:

1. Install [this build](https://drive.google.com/open?id=0B_IqTUgN98IXdWNKNHRtOF9UblU) of LootTweaker, also install CraftTweaker(I reproduced this issue with CT 1.10.2-3.0.15, but any 1.10.2 build should do). Any build of LootTweaker will reveal the issue, but the linked one has been tweaked to give a more useful error.
2. Launch Minecraft and open a world. 
3. Run the command /mt loottables all. There will be no errors in the log
4. Install EnderIO and EnderCore.
5. Repeat step 3. Errors will now occur.

____
#### Affected Versions (Do *not* use "latest"):

- CraftTweaker: 1.10.2-3.0.15
- EnderIO: 1.10.2-3.0.1.142_beta
- EnderCore: 1.10.2-0.4.1.61-beta
- Minecraft: 1.10.2
- Forge: 1.10.2-12.18.2.2134

#### Your most recent log file where the issue was present: 
https://paste.ee/p/TNh8M

### PR DESCRIPTION
**Pull request by Leviathan143**
See title. This fixes the issue discussed in #3927 
Loot weights will need to be adjusted as the loot weight system behaves differently to the previous probability based system. I tried to do this myself mathematically and failed due to differences between the two systems. The values will have to be adjusted by a process of guess and check, or something better if anyone knows of it.

A test world that can be used to easily test the loot tables can be found [here](https://drive.google.com/open?id=0B_IqTUgN98IXaVdjV0JtcFphZU0).
The world contains 4 buttons that each place a chest that uses one of the four loot tables modified by EnderIO.

### CODE REVIEW

File: src/main/java/crazypants/enderio/LootManager.java [hunk 1]
```diff
@@ -72,181 +61,99 @@ public void onLootTableLoad(LootTableLoadEvent evt) {
...
 
-  private LootItem createLootEntry(Item item, int minSize, int maxSize, double chance) {
-    return new LootItem(new ItemStack(item), chance, minSize, maxSize);
+  private LootEntry createLootEntry(Item item, int minSize, int maxSize, int chance, boolean enabled) {
+    return createLootEntry(item, 0, minSize, maxSize, chance, enabled);
   }
 
-  private LootItem createLootEntry(Item item, int ordinal, int minStackSize, int maxStackSize, double chance) {
-    return new LootItem(new ItemStack(item, 1, ordinal), chance, minStackSize, maxStackSize);
+  /**If enabled is true, an empty loot entry of the same weight is added.
```
Comment (HenryLoenwind) [inline src/main/java/crazypants/enderio/LootManager.java]: "If enabled is false"

**Comment by HenryLoenwind** [comment]
I'll need to look at this in my dev env later, it's a bit too large to really see in the diff view on github.

**Comment by Leviathan143** [comment]
After talking abou this on the MMD Discord with several people, MysteryM pointed out that I could use the RandomChance LootCondition instead of weights. I have done this, this implementation should behave similarly to the old system. It will also not require tweaking of loot values, though checking the loot spawn rates are still as intended is still a good idea. Empty loot entries are also no longer needed to preserve the loot spawn rates when loot is turned on/off in the config.

**Comment by HenryLoenwind** [comment]
nice.

**Pull request merged**